lib/src/piece/list.dart

```diff
@@ -55,8 +55,9 @@ class ListPiece extends Piece {
   /// The details of how this particular list should be formatted.
   final ListStyle _style;
 
-  ListPiece(this._before, this._elements, this._blanksAfter, this._after,
-      this._style) {
+  ListPiece(
+      this._before, this._elements, this._blanksAfter, this._after, this._style)
+      : assert(_elements.isNotEmpty) {
```

Comment:
[nit] This change probably deserves a doc change somewhere in this class. Consider adding a doc on the ctor, or (maybe better?) adding a note to the class level doc.

Reply:
Added a word on the class doc and a sentence on the constructor.